internal/pkg/vm/internal/vmcontext/syscalls.go

```diff
@@ -0,0 +1,92 @@
+package vmcontext
+
+import (
+	ffi "github.com/filecoin-project/filecoin-ffi"
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/proofs/verification"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/types"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/internal/runtime"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/crypto"
+	specsruntime "github.com/filecoin-project/specs-actors/actors/runtime"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/ipfs/go-cid"
+	"github.com/minio/blake2b-simd"
+)
+
+// Dragons: have the VM take a SysCalls object on construction (it will need a wrapper to charge gas)
+type syscalls struct {
+	gasTank *GasTracker
+}
+
+var _ specsruntime.Syscalls = (*syscalls)(nil)
+
+// VerifySignature implements Syscalls.
+func (sys syscalls) VerifySignature(signature crypto.Signature, signer address.Address, plaintext []byte) bool {
+	return types.IsValidSignature(plaintext, signer, signature)
+}
+
+// HashBlake2b implements Syscalls.
+func (sys syscalls) HashBlake2b(data []byte) [32]byte {
+	return blake2b.Sum256(data)
+}
+
+// ComputeUnsealedSectorCID implements Syscalls.
+func (sys syscalls) ComputeUnsealedSectorCID(sectorSize abi.SectorSize, pieces []abi.PieceInfo) (cid.Cid, error) {
+	panic("TODO")
+}
+
+// VerifySeal implements Syscalls.
+func (sys syscalls) VerifySeal(sectorSize abi.SectorSize, info abi.SealVerifyInfo) bool {
+	verifier := verification.NewFFIBackedProofVerifier()
+
+	commD := to32bytes(info.UnsealedCID.Bytes())
+	commR := to32bytes(info.OnChain.SealedCID.Bytes())
```

Comment:
Blocking: use the https://github.com/filecoin-project/go-fil-commcid library to convert CIDs to the raw commitments.